compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -1732,7 +1732,7 @@ object ProtobufGenerator {
   }
 
   val deprecatedAnnotation: String =
-    """@scala.deprecated(message="Marked as deprecated in proto file", "")"""
+    """@scala.annotation.nowarn("cat=deprecation") @scala.deprecated(message="Marked as deprecated in proto file", "")"""
```

Comment:
This should maybe just return a `Seq` now, but left as-is to reduce file changes

Reply:
nit: can it be made over two lines of output for readability?

Reply:
Actually, let's make it a Seq. IIRC, breaking it into two lines would mess up indentation.
 
Does adding the `nowarn` here works as expected? Will it prevent warnings in use sites both within generated code but enable them in user's code?

Reply:
returning a seq now in [`9561`](https://github.com/scalapb/ScalaPB/pull/1476/commits/9561bce82cfa3b3194020c0478212e25fd674e48) but there are other references to the field that need to be annotated still (see `Tests` section in the PR description)

Reply:
Q: why the definition site need the `nowarn` annotation? Shouldn't just the locations where it's referenced need the `nowarn` annotation?

Reply:
I believe both do, you can even see the warn message for the definition site:
```
[warn] /Users/mloyzer/workspace/mark/ScalaPB/examples/basic/target/scala-2.13/src_managed/main/scalapb/tutorial/addressbookv3/SomeTest.scala:10:73: value test in class SomeTest is deprecated: Marked as deprecated in proto file
[warn]     @scala.deprecated(message="Marked as deprecated in proto file", "") test: _root_.scala.Predef.String = "",
```

Reply:
as an update, the error:
```
[warn] /Users/mloyzer/workspace/mark/ScalaPB/examples/basic/target/scala-2.13/src_managed/main/scalapb/tutorial/addressbookv3/SomeTest.scala:9:18: value test in class SomeTest is deprecated: Marked as deprecated in proto file
[warn] final case class SomeTest(
```

means we'll need a `nowarn` at the class-level at which point it's going to silence everything under it (i.e all use sites as well), so i made a change to just apply the annotation there,  i want to add specs for that and then go onto OneOf and Enum